control-plane/connect-inject/common/openshift_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package common
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/constants"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestOpenShiftUID(t *testing.T) {
```

Comment:
I think you're just missing the case where the openshiftID is found but is empty